main.js

```diff
@@ -0,0 +1,94 @@
+// Clase Producto
+class Producto {
+  constructor(nombre, precio) {
+    this.nombre = nombre;
+    this.precio = precio;
+  }
+
+  calcularSubtotal(cantidad) {
+    return this.precio * cantidad;
+  }
+}
+
+// Clase Carrito
+class Carrito {
+  constructor() {
+    this.productos = [];
+  }
+
+  agregarProducto(producto, cantidad) {
```

Comment:
Si el usuario agrega el mismo producto dos veces, se crea una entrada duplicada en lugar de sumar la cantidad.Modificar agregarProducto para acumular si ya existe

```suggestion
 if (existente) {
    existente.cantidad += cantidad;
    existente.subtotal = existente.precio * existente.cantidad;
  }```